cohortextractor/tpp_backend.py

```diff
@@ -1019,6 +1019,103 @@ def _patients_with_events(
 
         return codelist_queries + extra_queries + [sql]
 
+    def patients_with_these_decision_support_values(self, **kwargs):
+        # First, we pop the kwargs.
+        algorithm = kwargs.pop("algorithm")
```

Comment:
Why use `kwargs` and `pop` here, rather than just accepting a fixed set of keyword arguments in the normal way?

**update**: Ah, possibly you've been looking at the `medications` and `clinical_events` methods which do this. But that's because they're slightly weird proxy methods which forward most of their arguments on to other functions. Most other functions use vanilla keyword args.

Reply:
Ah! That makes sense. I'd also forgotten how flexible Python is with `kwargs`. I'll change the method signature.

Reply:
Done. See `5695aff`.